src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -481,10 +494,8 @@ private static AgentArguments getAgentArguments(JenkinsRule r, String name) thro
         if (!launcher.getWorkDirSettings().isDisabled()) {
             commandLineArgs = launcher.getWorkDirSettings().toCommandLineArgs(c);
         }
-        File agentJar = new File(System.getProperty("java.io.tmpdir"), "agent.jar");
-        if (!agentJar.isFile()) {
-            FileUtils.copyURLToFile(new Slave.JnlpJar("agent.jar").getURL(), agentJar);
-        }
+        File agentJar = Files.createTempFile(Path.of(System.getProperty("java.io.tmpdir")), "agent", ".jar").toFile();
+        FileUtils.copyURLToFile(new Slave.JnlpJar("agent.jar").getURL(), agentJar);
```

Comment:
I'm guessing that this is to protect against a single Test having multiple threads that call start() on 2 different agents sumultaneously to avoid any synchronization around the file.